repose-aggregator/src/docs/asciidoc/services/datastores.adoc

```diff
@@ -3,4 +3,293 @@
+<5> The truststore used for validating the server this pool is connecting to.
+    This is typically set to the same path as the client authentication keystore.
+<6> The password for the client authentication truststore.
+    NOTE: This attribute is only used if the truststore-filename attribute is present.
+<7> Defines a list of hosts who has access to the distributed datastore API calls.
+    This does not add the host to the participating data storage nodes.
+<8> Setting the optional `allow-all` to `true` (defaults to `false`) will turn off host ACL checking.
+<9> Defines a host who has access to the distributed datastore API.
+<10> Defines the `port` on which all datastores in this `cluster` will listen.
+<11> Overrides the default `port`, defined for this `cluster` 's datasotores to listen, only for this particular `node`.
+
+=== Security
+The distributed datastore provides the option to encrypt communication between nodes using HTTP over SSL/TLS.
+As mentioned above, this is achieved by configuring a keystore in the `dist-datastore.cfg.xml` file.
+This feature can additionally provide client authentication during the SSL handshake.
+By both validating the client credentials and encrypting communication with the client, data in the datastore is made more secure.
+
+Assuming all Repose nodes are configured identically, the most straight-forward way to make use of this security would be to use a single unique keystore as both the keystore and the truststore.
+This can be achieved by not explicitly configuring a separate truststore.
+Since each datastore node will have a copy of the keystore, each node will trust every other node.
+
+Client authentication in SSL/TLS can act as as alternate form of client validation, performing a task similar to that of an access control list.
+As such, the usage of client authentication may replace the need to configure the allowed-hosts section of the `dist-datastore.cfg.xml` file.
+
+Note that distributed datastore will use a connection pool to communicate across nodes.
+If a connection pool is not configured, the default will be used.
+In nearly all cases, the connection pool being used should not be the default, but rather, a connection pool should be configured to use a keystore/truststore matching the keystore/trustore configured in the distributed datastore.
+That is, the distributed datastore may be thought of as a server, and clients in the connection pool as clients.
+Both the client and server need to be aware of how to communicate, and so they both must be configured with the appropriate secrets.
+
+For managing keystores and truststores, the aptly named keytool can be used.
+
+For more details, see:
+
+* http://www.eclipse.org/jetty/documentation/current/configuring-ssl.html#loading-keys-and-certificates
+* https://docs.oracle.com/javase/8/docs/technotes/tools/unix/keytool.html
+* https://docs.oracle.com/javase/8/docs/technotes/guides/security/crypto/CryptoSpec.html#KeystoreImplementation
```

Comment:
These links should probably be given shorter link text to make them more human-friendly.

For example:
`http://www.eclipse.org/jetty/documentation/current/configuring-ssl.html#loading-keys-and-certificates[Loading Keys and Certificates]`